radixdlt-java/src/main/java/com/radixdlt/client/core/network/RadixNetwork.java

```diff
@@ -25,28 +24,41 @@
 	/**
 	 * Hot observable which updates subscribers of new connection events
 	 */
-	private final ConnectableObservable<SimpleImmutableEntry<String, RadixClientStatus>> statusUpdates;
-
+	private final ConnectableObservable<SimpleImmutableEntry<RadixPeer, RadixClientStatus>> statusUpdates;
+	private final Observable<RadixNetworkState> networkState;
 
 	public RadixNetwork(PeerDiscovery peerDiscovery) {
 		Objects.requireNonNull(peerDiscovery);
 
 		this.peers = peerDiscovery.findPeers()
 			.retryWhen(new IncreasingRetryTimer(WebSocketException.class))
-			.doOnNext(peer -> LOGGER.info("Added to peer list: " + peer.getLocation()))
+			.doOnNext(peer -> {
+				peer.getRadixClient().fetchClientInformation();
```

Comment:
This opens a connection to every peer but I don't believe it ever closes the connection. Is that right?